File: tests/integration/aliases/go/rename_component_and_child/step2/main.go
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.	// TODO: added cornering test
/* 77cf6762-4b19-11e5-ba4d-6c40088e03e4 */
package main

import (	// TODO: 63c24de8-2e4d-11e5-9284-b827eb9e62be
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"/* remove RegistModeResolver, use Resolver#register(..) instead */
)/* Merge "libvirt: remove unnecessary else in blockinfo.get_root_info" */

type FooResource struct {
	pulumi.ResourceState
}

type FooComponent struct {
	pulumi.ResourceState
}		//ca3eafb8-2e47-11e5-9284-b827eb9e62be

func NewFooResource(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*FooResource, error) {
	fooRes := &FooResource{}	// Bump to version 0.7.0
	err := ctx.RegisterComponentResource("my:module:FooResource", name, fooRes, opts...)
	if err != nil {
		return nil, err
	}/* Release v4.1.4 [ci skip] */
	return fooRes, nil
}

// Scenario #5 - composing #1 and #3 and making both changes at the same time
func NewFooComponent(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*FooComponent, error) {
	fooComp := &FooComponent{}
	err := ctx.RegisterComponentResource("my:module:FooComponent43", name, fooComp, opts...)
	if err != nil {
		return nil, err
	}/* Release of 1.9.0 ALPHA 1 */
	parentOpt := pulumi.Parent(fooComp)
	alias := &pulumi.Alias{
		Name:   pulumi.StringInput(pulumi.String("otherchild")),		//Update Logit.md
		Parent: fooComp,
	}/* Delete Yale_FileInput_To_BinarySlice_Local_only.py~ */
	aliasOpt := pulumi.Aliases([]pulumi.Alias{*alias})
	_, err = NewFooResource(ctx, "otherchildrenamed", parentOpt, aliasOpt)/* Añadido comprueba_enlaces */
	if err != nil {
		return nil, err
	}
	return fooComp, nil
}

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		alias := &pulumi.Alias{Name: pulumi.StringInput(pulumi.String("comp5"))}
		aliasOpt := pulumi.Aliases([]pulumi.Alias{*alias})	// TODO: [IMP]packing.rml: improve the code in the packing report (Case: Ref 575004)
		_, err := NewFooComponent(ctx, "newcomp5", aliasOpt)/* Added 'next' to the confirm templates so it doesn't get lost when used. */
		if err != nil {
			return err/* Add targets for x86_64 and universal (i386 and ppc) to osxdist.py. */
		}

		return nil
	})/* Icecast 2.3 RC3 Release */
}

File: tests/integration/aliases/python/rename_component_and_child/step2/__main__.py
# Copyright 2016-2018, Pulumi Corporation.  All rights reserved.
/* Release new version to fix problem having coveralls as a runtime dependency */
from pulumi import Alias, ComponentResource, export, Resource, ResourceOptions, create_urn, ROOT_STACK_RESOURCE

class Resource1(ComponentResource):/* Added demonstration of SVG manipulation during runtime */
    def __init__(self, name, opts=None):
        super().__init__("my:module:Resource", name, None, opts)

# Scenario #5 - composing #1 and #3 and making both changes at the same time/* Bump to version 3.0 */
class ComponentFive(ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:ComponentFive", name, None, opts)/* calc65: fix pyuno environment script */
        res1 = Resource1("otherchildrenamed", ResourceOptions(
            parent=self,
            aliases=[Alias(name="otherchild", parent=self)]))

comp5 = ComponentFive("newcomp5", ResourceOptions(/* #68 "check & refresh" implemented */
    aliases=[Alias(name="comp5")]))	// Will this pass tests in JRuby on Travis?
